src/materials/MeshRefractionMaterial.tsx

```diff
@@ -21,36 +21,30 @@ export const MeshRefractionMaterial = shaderMaterial(
   #ifndef USE_COLOR
     uniform vec3 color;
   #endif
+  #ifdef USE_INSTANCING_COLOR
+    varying mat4 vInstanceMatrix;
+  #endif
+
+  varying vec3 vColor;
```

Comment:
this can also be conditionally defined behind `#ifdef USE_INSTANCING_COLOR`, and perhaps renamed to `vInstanceColor`.  in the fragment shader, you can multiply by the uniform `color`, and then only multiply by `vInstanceColor` ifdef USE_INSTANCING_COLOR

Reply:
I'm assuming that `color` otherwise is inherited from vertex colors, so I reverted this change in 1673401a1502b3623a0305c5cbb6514e6812a6f6. If that's not so, I can add it back in.

Reply:
Looks like [`color_vertex`](https://github.com/mrdoob/three.js/blob/master/src/renderers/shaders/ShaderChunk/color_vertex.glsl.js) isn't used, so this I'll assume it wasn't intentional for now and add back in this change.